### PR TITLE
Initialization code moved to tests

### PR DESCRIPTION
**Pull request by gastonfournier**
This moves the setup to the test so it follows the same approach as https://github.com/Unleash/sdk-integration-tester/blob/bf4fb8b9e3f6cca6c7ca82534efadd7fae790c3d/src/client-spec.test.ts

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ services:
       DATABASE_USERNAME: unleash_user
       DATABASE_PASSWORD: unleash.the.password
       DATABASE_SSL: false
-      INIT_CLIENT_API_TOKENS: "test-server:development.unleash-insecure-api-token"
+      INIT_CLIENT_API_TOKENS: "*:development.unleash-insecure-api-token"
```
Comment (gastonfournier) [inline docker-compose.yml:38]: Maybe issue reported: https://linear.app/unleash/issue/2-262/race-condition-issue-with-token-initialization

File: docker-compose.yml [hunk 2]
```diff
@@ -46,15 +46,6 @@ services:
...
 
-  initializer:
-    image: alpine/curl
-    depends_on:
-      - unleash-server
-    volumes:
-       - ./server/initial-state.json:/initial-state.json
-       - ./server/init.sh:/init.sh
-    command: /init.sh
-
```
Comment (gastonfournier) [inline docker-compose.yml:57]: Not needed, the test case takes care of initializing state

File: java/src/main/java/io/getunleash/sdktester/UnleashSdkTesterConfig.java [hunk 3]
```diff
@@ -27,6 +27,7 @@ public UnleashConfig unleashConfig() {
                 .unleashAPI(unleashUrl)
                 .customHttpHeader("Authorization", unleashApiKey)
                 .synchronousFetchOnInitialisation(true)
+                .fetchTogglesInterval(1)
```
Comment (gastonfournier) [inline java/src/main/java/io/getunleash/sdktester/UnleashSdkTesterConfig.java:30]: This is in seconds, so it's the lower refresh interval we can have

File: node/index.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ const { Unleash } = require('unleash-client')
 const unleash = new Unleash({
   appName: 'node-test-server',
   url,
-  refreshInterval: 2000,
+  refreshInterval: 500,
```
Comment (gastonfournier) [inline node/index.js:15]: This one is in milliseconds

File: python/main.py [hunk 5]
```diff
@@ -14,6 +14,7 @@
     url=API_URL,
     app_name="python-test-server",
     custom_headers={"Authorization": API_KEY},
+    refresh_interval = 1
```
Comment (gastonfournier) [inline python/main.py:17]: This one is in seconds

File: run-tests.sh [hunk 6]
```diff
@@ -1,13 +1,16 @@
...
     SDK=( $i )
-    ./server/run-test.sh ${SDK[0]} ${SDK[1]} ${SDK[2]}
+    docker compose up -d postgres unleash-server
+
+    ./wait-unleash-server-has-token.sh || exit 1
+    echo "Now server is setup"
+
+    ./run-test.sh ${SDK[0]} ${SDK[1]} ${SDK[2]}
+    docker compose stop postgres unleash-server
+    docker compose rm -f postgres unleash-server
```
Comment (gastonfournier) [inline run-tests.sh:14]: On every iteration, we start with a clean database and a clean server, this negatively impacts execution time of the tests but increases confidence

File: src/client-spec.test.ts [hunk 7]
```diff
@@ -107,6 +111,7 @@ specs
           }
         )
         expect(statusCode).toBe(202)
+        await sleep(1200);
```
Comment (gastonfournier) [inline src/client-spec.test.ts:114]: Wait at least the minimum refresh interval plus a threshold considering latency.

File: src/simple.test.ts [hunk 8]
```diff
@@ -34,6 +41,22 @@ describe.each([
...
+      {
+        headers: {
+          Authorization: UNLEASH_TOKEN
+        },
+        json: state
+      }
+    )
+    expect(statusCode).toBe(202)
+    await sleep(5000);
+  })
```
Comment (gastonfournier) [inline src/simple.test.ts:58]: Now the test takes care of the state and also waits allowing the SDKs to refresh

File: src/simple.test.ts [hunk 9]
```diff
@@ -34,6 +42,22 @@ describe.each([
...
+      `${UNLEASH_URL}/api/admin/state/import`,
+      {
+        headers: {
+          Authorization: UNLEASH_TOKEN
+        },
+        json: state
+      }
+    )
+    expect(statusCode).toBe(202)
+    await sleep(WAIT_SDK_REFRESH_MS);
```
Comment (nunogois) [inline src/simple.test.ts:58]: Maybe use a constant for `client-spec.test.ts` as well.

**Review by nunogois — APPROVED** [review]
LGTM - Naturally tests take a bit longer but I no longer have those failing tests on the first run.
Would be great if we could make the SDK implementations consistent with a MS refresh_interval in all cases. I'm sure this project will help us find a lot of these details that we may want to change in future versions of the SDKs.

**Review by chriswk — APPROVED** [review]

**Pull request closed without merging**